packages/yarnpkg-shell/sources/index.ts

```diff
@@ -52,6 +52,50 @@ export type ShellState = {
   variables: {[key: string]: string},
 };
 
+enum StreamType {
+  Readable = 0b01,
+  Writable = 0b10,
```

Comment:
Can it be both `Readable` and `Readable`?

Reply:
(You probably meant both `Readable` and `Writable`) Technically it can be both, but practically we don't have a use for that. (The current implementation works as expected if I pass `StreamType.Readable | StreamType.Writable` to the function).